src/ngRoute/route.js

```diff
@@ -160,7 +171,7 @@ function $RouteProvider() {
    */
   this.when = function(path, route) {
     //copy original route object to preserve params inherited from proto chain
-    var routeCopy = angular.copy(route);
+    var routeCopy = createShallowCopy(route);
```

Comment:
can't you use this one: https://github.com/angular/angular.js/blob/3ba29c464da9d73de0ea7c7716dc1ecc60bc5287/src/Angular.js#L941 ?